Src/Base/AMReX_GpuAtomic.H

```diff
@@ -22,13 +22,13 @@ namespace detail {
 
     template <typename R, typename I, typename F>
     AMREX_GPU_DEVICE AMREX_FORCE_INLINE
-    R atomic_op (R* address, R val, F f) noexcept
+    R atomic_op (R* AMREX_RESTRICT const address, R const val, F const f) noexcept
     {
 #if defined(__SYCL_DEVICE_ONLY__)
         constexpr auto mo = sycl::memory_order::relaxed;
         constexpr auto as = sycl::access::address_space::global_space;
         static_assert(sizeof(R) == sizeof(I), "sizeof R != sizeof I");
-        I* add_as_I = reinterpret_cast<I*>(address);
+        I* AMREX_RESTRICT add_as_I = reinterpret_cast<I*>(address);
```

Comment:
since we handle more than one pointer in this kernel, I was not sure if it might be useful to add the restrict here to be very explicit.